pkg/gateway/middleware/tenant.go

```diff
@@ -5,26 +5,39 @@ import (
 	"net/http"
 
 	"github.com/gorilla/mux"
+	"github.com/skygeario/skygear-server/pkg/gateway/db"
 	"github.com/skygeario/skygear-server/pkg/gateway/model"
+	"github.com/skygeario/skygear-server/pkg/gateway/logging"
 	"github.com/skygeario/skygear-server/pkg/core/config"
 )
 
 type TenantMiddleware struct {
+	DBOpener db.DBOpener
```

Comment:
I much prefer using `sqlx.DB` or the wrapper `db.Conn` here, because the db connection pool should be alive along side with the gateway server.